src/transformers/models/omnivore/modeling_omnivore.py

```diff
@@ -0,0 +1,987 @@
+
+
+class OmnivoreImageHead(nn.Module):
+    def __init__(self, in_features=1024, out_features=1000, bias=True):
+        super().__init__()
+        self.image_head = nn.Linear(in_features, out_features, bias)
+
+    def forward(self, hidden_state):
+        logits = self.image_head(hidden_state)
+        return logits
+
+
+class OmnivoreVideoHead(nn.Module):
+    def __init__(self, in_features=1024, out_features=400, bias=True):
+        super().__init__()
+        self.video_head = nn.Linear(in_features, out_features, bias)
+        self.dropout = nn.Dropout(p=0.5)
+
+    def forward(self, hidden_state):
+        logits = self.video_head(hidden_state)
+        logits = self.dropout(logits)
+        return logits
+
+
+class OmnivoreRGBDHead(nn.Module):
+    def __init__(self, in_features=1024, out_features=19, bias=True):
+        super().__init__()
+        self.rgbd_head = nn.Linear(in_features, out_features, bias)
+
+    def forward(self, hidden_state):
+        logits = self.rgbd_head(hidden_state)
+        return logits
+
+
+class OmnivorePreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = OmnivoreConfig
+    base_model_prefix = "omnivore"
+    main_input_name = "pixel_values"
+    supports_gradient_checkpointing = True
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, (nn.Linear, nn.Conv3d)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, OmnivoreSwinBackbone):
+            module.gradient_checkpointing = value
+
+
+OMNIVORE_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`OmnivoreConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+OMNIVORE_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
```

Comment:
```suggestion
        pixel_values (`torch.FloatTensor` of shape `(batch_size, time, num_channels, height, width)`):
```
To me it makes most sense to let the model accept tensors of this shape (whether to call it "time", "depth" or "num_frames" is another discussion). 

It seems that the original implementation uses (batch_size, num_channels, time, height, width). Was wondering why.